### PR TITLE
MMCA-4785 | Upgrade Scala & Libraries

### PR DESCRIPTION
**Pull request by Neozxc**
**Acceptance criteria:** 
- Ensure the scala version upgrades to 3.3.4 and library dependencies to the latest version.

**Dev Tasks:**
- [x] Update scala version to 3.3.4
- [x] Add: "-Wconf:src=routes/.*:s", "-Wconf:msg=Flag.*repeatedly:s" in build.sbt
- [x] Update the dependencies
- [x] Make sure services have no compilation issues
- [x] Make sure unit tests pass and the build job is green

**FYI:**
Currently while compiling, we will encounter the _warning_ below, more information can be found in the [ticket comments](https://jira.tools.tax.service.gov.uk/browse/MMCA-4785).

```scala
[warn] -- Deprecation Warning: customs-guarantee-account-frontend/app/controllers/actions/IdentifierAction.scala:47:64 
[warn] 47 |    authorised().retrieve(Retrievals.credentials and Retrievals.name and Retrievals.email
[warn]    |                                                     ^^^^^^^^^^^^^^^
[warn]    |value name in trait Retrievals is deprecated since 8.2.0: See Tech Blog Post dated 2024-02-26.
[warn] one warning found
```

### CODE REVIEW

File: app/connectors/CustomsFinancialsApiConnector.scala [hunk 1]
```diff
@@ -86,19 +93,20 @@ class CustomsFinancialsApiConnector @Inject()(httpClient: HttpClient,
...
-                                                    gan: String,
-                                                    onlyOpenItems: Boolean,
-                                                    from: LocalDate,
-                                                    to: LocalDate)(
-                                                    implicit hc: HeaderCarrier): Future[Either[GuaranteeResponses, Seq[GuaranteeTransaction]]] = {
+  def retrieveRequestedGuaranteeTransactionsDetail(gan: String,
+                                                   onlyOpenItems: Boolean,
+                                                   from: LocalDate,
+                                                   to: LocalDate)
+                                                  (implicit hc: HeaderCarrier): Future[Either[GuaranteeResponses, Seq[GuaranteeTransaction]]] = {
```
Comment (AdamCoops) [inline app/connectors/CustomsFinancialsApiConnector.scala:100]: This line looks like it's over the length limit to me

**Comment by AdamCoops** [comment]
as mentioned in the pull request there is a value name in trait Retrievals is deprecated warning. It's been marked as to be addressed in another ticket.

File: build.sbt [hunk 2]
```diff
@@ -33,28 +32,27 @@ lazy val microservice = Project(appName, file("."))
...
-      "-Wunused:imports", "-Wunused:params",
-      "-Wunused:patvars", "-Wunused:implicits",
-      "-Wunused:explicits", "-Wunused:privates",
-      "-Wconf:cat=unused-imports&src=routes/.*:s"),
+    scalacOptions := scalacOptions.value
+      .diff(Seq("-Wunused:all")) ++ Seq("-Wconf:src=routes/.*:s", "-Wconf:msg=Flag.*repeatedly:s"),
 
-    Test / scalacOptions ++= Seq("-Wunused:imports", "-Wunused:params",
-      "-Wunused:patvars", "-Wunused:implicits", "-Wunused:explicits", "-Wunused:privates"),
+    Test / scalacOptions ++= Seq(
```
Comment (AdamCoops) [inline build.sbt:46]: are all these scalacOptions needed here? we are removing most of them from the non-tests.

File: test/controllers/RequestedTransactionsControllerSpec.scala [hunk 3]
```diff
@@ -133,19 +140,16 @@ class RequestedTransactionsControllerSpec extends SpecBase {
...
-        BigDecimal(limit),
-        BigDecimal(balance)
-      )))
+    val guaranteeAccount: GuaranteeAccount = GuaranteeAccount(
+      someGan, eori, AccountStatusOpen, Some(GeneralGuaranteeBalance(BigDecimal(limit), BigDecimal(balance))))
 
-    val amt = Amounts("20.00", Some("30.00"), Some("10.00"), "2020-08-01")
-    val tt = TaxType("VAT", amt)
-    val ttg = TaxTypeGroup(taxTypeGroup = "VAT", amounts = amt, taxType = tt)
+    val amt: Amounts = Amounts("20.00", Some("30.00"), Some("10.00"), "2020-08-01")
```
[the remark this answers is not in the document]
Comment (Neozxc) [reply test/controllers/RequestedTransactionsControllerSpec.scala:148]: For the testing options we still keep them, as they have been kept in other repos as well shown [here](https://github.com/hmrc/customs-duty-deferment-frontend/blob/e097b8bd957e44dcd853cb3c62515a64b64715b4/build.sbt#L25) and [here](https://github.com/hmrc/customs-cash-account-frontend/blob/879f01c789f65cfaf60c5ddf09cf12798279d743/build.sbt#L53), but yes we have removed them from non-testing options

File: build.sbt (hunk 2 above)
Comment (Neozxc) [reply build.sbt:46]: For the testing options we still keep them, as they have been kept in other repos as well shown [here](https://github.com/hmrc/customs-duty-deferment-frontend/blob/e097b8bd957e44dcd853cb3c62515a64b64715b4/build.sbt#L25) and [here](https://github.com/hmrc/customs-cash-account-frontend/blob/879f01c789f65cfaf60c5ddf09cf12798279d743/build.sbt#L53), but yes we have removed them from non-testing options

File: app/connectors/CustomsFinancialsApiConnector.scala (hunk 1 above)
Comment (Neozxc) [reply app/connectors/CustomsFinancialsApiConnector.scala]: Addressed in this [commit](https://github.com/hmrc/customs-guarantee-account-frontend/pull/86/commits/c169bbfb4c8efe54a4e6ea4b455973a6624ba249)

File: test/controllers/RequestedTransactionsControllerSpec.scala (hunk 3 above)
[the remark this answers is not in the document]
Comment (Neozxc) [reply test/controllers/RequestedTransactionsControllerSpec.scala]: Addressed in this [commit](https://github.com/hmrc/customs-guarantee-account-frontend/pull/86/commits/c169bbfb4c8efe54a4e6ea4b455973a6624ba249)

**Review by AdamCoops — APPROVED** [review]

**Pull request merged**